panels/info/cc-info-panel.c

```diff
@@ -1798,6 +1817,72 @@ info_panel_setup_overview (CcInfoPanel  *self)
   g_clear_pointer (&os_info, g_hash_table_destroy);
 }
 
+static gboolean
+read_config_file (CcInfoPanel *self)
+{
+  GKeyFile *keyfile = NULL;
+  GError *error = NULL;
+  gchar **groups = NULL;
+  gboolean ret = FALSE;
+  gsize n_groups;
+  int i;
+
+  if (!g_file_test (CONFIG_FILEPATH, G_FILE_TEST_EXISTS))
+    return FALSE;
+
+  keyfile = g_key_file_new ();
+  if (!g_key_file_load_from_file (keyfile, CONFIG_FILEPATH, G_KEY_FILE_NONE, &error))
+    {
+      g_warning ("Could not read file %s: %s\n", CONFIG_FILEPATH, error->message);
+      g_error_free (error);
+      goto out;
+    }
+
+  groups = g_key_file_get_groups (keyfile, &n_groups);
+  for (i = 0; i < n_groups; i++)
+    {
+      size_t len;
+
+      /* We only support "Info" group here */
+      if (g_strcmp0 (groups[i], "Info") != 0)
+        continue;
+
+      self->priv->branding_logo_path = g_key_file_get_string (keyfile, groups[i], "logo", NULL);
+      if (self->priv->branding_logo_path == NULL) {
+        g_warning ("Could not find 'logo' in %s\n", CONFIG_FILEPATH);
+        goto out;
+      }
+
+      break;
+    }
+
+  ret = TRUE;
+
+ out:
+  g_strfreev (groups);
+  g_key_file_free (keyfile);
+
+  return ret;
+}
+
+static void
+update_branding_specific_info (CcInfoPanel *self)
+{
+  if (!read_config_file (self)) {
```

Comment:
I would change `read_config_file()` to not return any value, and move the debug message inside the next if. Basically it does not make sense to check for both the function returning false and the logo path being NULL.